### PR TITLE
Bug 1921909: vendor in updated openshift/docker-distribution

### PR DESCRIPTION
**Pull request by joelddiaz**
This allows setting REGISTRY_STORAGE_S3_CREDENTIALSCONFIGPATH to point to a file containing AWS credentials (instead of setting environment variables containing the credentials).

The registry operator can now chose to mount a Secret/ConfigMap containing an AWS credentials configuration file which should allow using AWS authentication methods besides the (still supported) AWS access and secret keys.

### CODE REVIEW

**Comment by joelddiaz** [comment]
this depends on https://github.com/openshift/docker-distribution/pull/23 and will have to update the vendoring before this PR can merge
/hold

**Comment by joelddiaz** [comment]
/unhold
openshift/docker-distribution#23 merged, and this PR has been updated to vendor in the real branch

**Review by ricardomaraschini — APPROVED** [review]
/approve

**Comment by ricardomaraschini** [comment]
@joelddiaz I am working into getting some debug around this failing test. It got introduced recently.

**Comment by sdodson** [comment]
/retitle Bug 1921909: vendor in updated openshift/docker-distribution

**Pull request merged**
